### PR TITLE
fix: Context.startForegroundService did not then call Service.startFo…

### LINKED ISSUE #1104: Context.startForegroundService did not then call Service.startForeground
1. What problem do you get?
`Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{ad3cfbc u0 <application.package.name>/com.liulishuo.filedownloader.services.FileDownloadService$SharedMainProcessService}
`
2. Which version of FileDownloader are you using when you produce such problem?
1.7.5

3. How to reproduce such problem?
When `OS >= Oreo` and App is still in foreground

4. Do you set `FileDownloadLog.NEED_LOG=true`?
Yes

I think problem here is that in `FileDownloadService` class you check for `needMakeServiceForeground` with conditions `Build.VERSION.SDK_INT >= Build.VERSION_CODES.O` and `!isAppOnForeground()` However, if `OS >= Oreo` then it should always call `makeServiceForeground()` doesn't matter if app is in foreground or not.

Have a look here https://developer.android.com/about/versions/oreo/background 

If you think that's correct then I can open PR

### PR DESCRIPTION
**Pull request by rantianhua**
…reground

Closes #1104

### CODE REVIEW

File: library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadService.java [hunk 1]
```diff
@@ -72,18 +72,25 @@ public void onCreate() {
...
+    }
+
     private void makeServiceForeground() {
         ForegroundServiceConfig config =
                 CustomComponentHolder.getImpl().getForegroundConfigInstance();
         if (FileDownloadLog.NEED_LOG) {
             FileDownloadLog.d(this, "make service foreground: %s", config);
         }
-        if (config.isNeedRecreateChannelId()) {
+        if (config.isNeedRecreateChannelId() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```
[the remark this answers is not in the document]
Comment (rantianhua) [reply library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadService.java]: There is no this api under Android O

[the remark this answers is not in the document]
Comment (Jacksgong) [reply library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadService.java]: Which API?

[the remark this answers is not in the document]
Comment (rantianhua) [reply library/src/main/java/com/liulishuo/filedownloader/services/FileDownloadService.java]: createNotificationChannel

File: library/src/main/java/com/liulishuo/filedownloader/model/ServiceStatusModel.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+package com.liulishuo.filedownloader.model;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+public class ServiceStatusModel implements Parcelable {
+
+    private boolean isHandled;
```
Comment (Jacksgong) [inline library/src/main/java/com/liulishuo/filedownloader/model/ServiceStatusModel.java]: I think `isHandled` is useless, if we bind on the several clients, we also can't know the first one is the right one, and bind on the several clients also not support and not be allowed on FileDownloader.

File: library/src/main/java/com/liulishuo/filedownloader/services/FDServiceSeparateHandler.java [hunk 3]
```diff
@@ -163,6 +164,28 @@ public void onDestroy() {
         MessageSnapshotFlow.getImpl().setReceiver(null);
     }
 
+    @Override
+    public void checkRunServiceForeground(ServiceStatusModel serviceStatusModel) {
+        final int n = callbackList.beginBroadcast();
```
[the remark this answers is not in the document]
Comment (Jacksgong) [reply library/src/main/java/com/liulishuo/filedownloader/services/FDServiceSeparateHandler.java]: that is a point, please check again with service lifecycle:

image

[the remark this answers is not in the document]
Comment (rantianhua) [reply library/src/main/java/com/liulishuo/filedownloader/services/FDServiceSeparateHandler.java]: updated

**Pull request merged**